### ISSUE TITLE
ALTER TYPE - RENAME VALUE

### ISSUE THREAD

**Issue by adomokos** [sqlfluff/sqlfluff#4075]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

```
% sqlfluff --version
sqlfluff, version 1.4.2
```

I tried to lint this statement:

```sql
ALTER TYPE financial.reporting_statuses RENAME VALUE 'partially' TO 'partially-reported';
```

Ran this command:

```
sqlfluff lint --dialect postgres test.sql
```

Received this error:

```
== [test.sql] FAIL
L:   5 | P:   1 |  PRS | Line 5, Position 1: Found unparsable section: 'ALTER
                       | TYPE audit_core.liquidation_status...'
WARNING: Parsing errors found and dialect is set to 'postgres'. Have you configured your dialect correctly?
```

### Expected Behaviour

It should not find `sqlfluff` the ALTER TYPE statement unparsable.

### Observed Behaviour

Unparsable error reported.

### How to reproduce

```
% cat test.sql
ALTER TYPE financial.reporting_statuses RENAME VALUE 'partially' TO 'partially-reported';
```

Run:

```
% sqlfluff lint --dialect postgres test.sql
```

Error reported:

```
Line 1, Position 1: Found unparsable section: 'ALTER
                       | TYPE financial.reporting_statuses ...
```

### Dialect

postgres

### Version

sqlfluff, version 1.4.2

### Configuration

```
[sqlfluff]
verbose = 1
dialect = postgres
exclude_rules = L031,L011,L030,L059

[sqlfluff:rules]
max_line_length = 100
```

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)
